templates/app/controllers/orders_controller.rb

```diff
@@ -6,80 +6,19 @@ class OrdersController < StoreController
   respond_to :html
 
   before_action :store_guest_token
-  before_action :assign_order, only: :update
-  # note: do not lock the #edit action because that's where we redirect when we fail to acquire a lock
-  around_action :lock_order, only: :update
 
   def show
     @order = Spree::Order.find_by!(number: params[:id])
     authorize! :show, @order, cookies.signed[:guest_token]
   end
 
-  def update
-    authorize! :update, @order, cookies.signed[:guest_token]
-    if @order.contents.update_cart(order_params)
-      @order.next if params.key?(:checkout) && @order.cart?
-
-      respond_with(@order) do |format|
-        format.html do
-          if params.key?(:checkout)
-            redirect_to checkout_state_path(@order.checkout_steps.first)
-          else
-            redirect_to cart_path
-          end
-        end
-      end
-    else
-      respond_with(@order)
-    end
-  end
-
-  # Shows the current incomplete order from the session
-  def edit
-    @order = current_order(build_order_if_necessary: true)
-    authorize! :edit, @order, cookies.signed[:guest_token]
-    if params[:id] && @order.number != params[:id]
-      flash[:error] = t('spree.cannot_edit_orders')
-      redirect_to cart_path
-    end
-  end
-
-  def empty
-    if @order = current_order
-      authorize! :update, @order, cookies.signed[:guest_token]
-      @order.empty!
-    end
-
-    redirect_to cart_path
-  end
+  private
 
   def accurate_title
-    if @order && @order.completed?
-      t('spree.order_number', number: @order.number)
```

Comment:
The new OrdersController only has the `show` action, so we can be sure that the order exists and has been completed.